dev/MANUALLY_GENERATING_IMAGE_CACHE_AND_CONSTRAINTS.md

```diff
@@ -73,6 +74,89 @@ git commit -m "Your commit message here" --no-verify
 git push
 ```
 
+# Manually updating already tagged constraint files
+
+Sometimes - very rarely - we need to fix historical constraint files when Airflow fails to install with the
+constraints that were used in the past. This happened already several times and usually only happens when
+there is a backwards-incompatible change in the build environment in Python installation toolchain
+(pip, setuptools, wheel, Cython etc.). The Python build environment is not controllable by us - by default
+pip uses `build isolation` which means that it will install the latest version of the build tools that
+are matching the expectations of the package being build and it might mean that new version of such tools can
```

Comment:
nit: `expectations of the package being *built* and it might mean that new version of such tools can`